lsstseries/analysis/structurefunction2.py

```diff
@@ -41,20 +55,48 @@ def calc_sf2(time, flux, err, bins, band, band_to_calc=None):
 
     assert hasattr(band_to_calc, "__iter__") is True
 
-    sf2 = {}
+    ids = []
+    dts = []
+    bands = []
+    sf2s = []
     for b in band_to_calc:
```

Comment:
[Optional] I know that this is existing code, but it took me a moment to figure out what was going on here. These two lines are effectively a set intersection. I think the following logic is the same, and is a little more readable:

```
band_intersection = set(np.unique(band))
if band_to_calc:
  band_intersection = band_intersection.intersection(band_to_calc) // this is ok with set or single string

for b in band_intersection:
  // do stuff
```

Reply:
Nice, this streamlines it quite a bit. Implemented

Reply:
Err, reverted back. This was causing the CI to fail. Not entirely sure why, perhaps the fact that sets in Python are not ordered? Don't want this to hold things up.